app/src/main/java/com/example/capsule/ui/itemDetails/ItemDetailsFragment.kt

```diff
@@ -117,8 +140,27 @@ class ItemDetailsFragment : Fragment() {
         var category = categorySpinner.selectedItem.toString()
         var material = materialSpinner.selectedItem.toString()
         var season = seasonSpinner.selectedItem.toString()
-        var price = priceEditText.text.toString()
+        var price = "0.00"
+        if (!priceEditText.text.toString().isEmpty()) {
+            price = String.format("%.2f", priceEditText.text.toString().toDouble())
+        }
         var purchaseLocation = purchaseLocationSpinner.selectedItem.toString()
-
+        var clothingEntry = Clothing(
+            name = itemName,
+            category = category,
+            material = material,
+            season = season,
+            price = price.toDouble(),
+            purchase_location = purchaseLocation,
+            img_uri = imgUriString
+        )
+        itemDetailsViewModel.insert(clothingEntry)
+        val nextFrag: Fragment? = ClosetFragment()
+        if (nextFrag != null) {
+            requireActivity().supportFragmentManager.beginTransaction()
+                .replace(R.id.nav_host_fragment_activity_main, nextFrag, R.string.fragment_key.toString())
```

Comment:
The R.string.fragment_key might have to be unique because it is used to retrieve the fragment if necessary. Right now its used for both the ItemDetailsFragment and the ClosetFragment. Not sure if this might a problem later. If this fragment does not need to be retrieved I think you don't need this parameter, otherwise might be safe to use unique names?